plugins/defaultplugins/srplugin/vppcalls/srv6_test.go

```diff
@@ -0,0 +1,930 @@
+				},
+			},
+		},
+		{
+			Name:          "missing interface in swIndexes (addition with endDX6 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX6: &srv6.LocalSID_EndDX6{
+					NextHop:           nextHop.String(),
+					OutgoingInterface: ifaceBOutOfidxs,
+				},
+			},
+		},
+		{
+			Name:          "invalid IP address (addition with endDX6 behaviour)",
+			ExpectFailure: true,
+			Input: &srv6.LocalSID{
+				FibTableID: 10,
+				EndFunctionDX6: &srv6.LocalSID_EndDX6{
+					NextHop:           invalidIPAddress,
+					OutgoingInterface: ifaceA,
+				},
+			},
+		},
+	}
+
+	// Run all cases
+	for _, td := range cases {
+		t.Run(td.Name, func(t *testing.T) {
+			func() { //wrapping in another function to properly teardown things inside deferred function in case of assertion failure (i.e. connection)
+				ctx := vppcallmock.SetupTestCtx(t)
+				defer ctx.TeardownTestCtx()
+				// prepare reply
+				if td.FailInVPP {
+					ctx.MockVpp.MockReply(&sr.SrLocalsidAddDelReply{Retval: 1})
+				} else {
+					ctx.MockVpp.MockReply(&sr.SrLocalsidAddDelReply{})
+				}
+				// make the call
+				err := vppcalls.NewSRv6Calls().AddLocalSid(sidA, td.Input, swIfIndex, logrus.DefaultLogger(), ctx.MockChannel, nil)
+				// verify result
+				if td.ExpectFailure {
+					Expect(err).Should(HaveOccurred())
+				} else {
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(ctx.MockChannel.Msg).To(Equal(td.Expected))
+				}
+			}()
+		})
+	}
+}
+
+// TestDeleteLocalSID tests all cases for method DeleteLocalSID
+func TestDeleteLocalSID(t *testing.T) {
+	// Prepare different cases
+	cases := []struct {
+		Name      string
+		Fail      bool
+		Sid       net.IP
+		MockReply govppapi.Message
+		Verify    func(error, govppapi.Message)
+	}{
+		{
+			Name:      "simple delete of local sid",
+			Sid:       sidA,
+			MockReply: &sr.SrLocalsidAddDelReply{},
+			Verify: func(err error, catchedMsg govppapi.Message) {
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(catchedMsg).To(Equal(&sr.SrLocalsidAddDel{
+					IsDel:        1,
+					LocalsidAddr: sidA,
+				}))
+			},
+		},
+		{
+			Name:      "failure propagation from VPP",
+			Sid:       sidA,
+			MockReply: &sr.SrLocalsidAddDelReply{Retval: 1},
+			Verify: func(err error, msg govppapi.Message) {
+				Expect(err).Should(HaveOccurred())
+			},
+		},
+	}
+
+	// Run all cases
+	for _, td := range cases {
+		t.Run(td.Name, func(t *testing.T) {
+			func() { //wrapping in another function to properly teardown things inside deferred function in case of assertion failure (i.e. connection)
+				ctx := vppcallmock.SetupTestCtx(t)
+				defer ctx.TeardownTestCtx()
+				// data and prepare case
+				calls := vppcalls.NewSRv6Calls()
+				localsid := &srv6.LocalSID{
+					FibTableID: 10,
+					BaseEndFunction: &srv6.LocalSID_End{
+						Psp: true,
+					},
+				}
+				calls.AddLocalSid(td.Sid, localsid, swIfIndex, logrus.DefaultLogger(), ctx.MockChannel, nil)
```

Comment:
Shouldn't this be checked for error as well?

Reply:
Nope. AddLocalSid is tested by TestAddLocalSid function and i don't want to retest it here. It is part of setup and if it fails, then also test will fail because it is depending on correct setup.